tests/cephadm/test_custom_container.py

```diff
@@ -0,0 +1,289 @@
+from json import loads
+
+from ceph.waiter import WaitUntil
+from cli.cephadm.cephadm import CephAdm
+from utility.utils import Log, get_cephci_config
+
+log = Log(__name__)
+
+PRIVATE_REG_USER = "myregistryusername"
+PRIVATE_REG_PASS = "myregistrypassword1"
+OPT_AUTH = "/opt/registry/auth"
+OPT_CERTS = "/opt/registry/certs"
+OPT_DATA = "/opt/registry/data"
+
+
+class ClusterCustomContainerError(Exception):
+    pass
+
+
+def update_cert(ceph):
+    """Check if the certificate is updated on all other nodes in cluster
+    Args:
+        ceph(ceph.ceph.Ceph): CephNode or list of CephNode object
+    """
+
+    out, _ = ceph.exec_command(cmd=f"sudo trust list | grep -i {ceph.hostname}")
+    if ceph.hostname in out:
+        log.info("Certificate added to network node successfully")
+    else:
+        log.error("Certificate failed to add in network node")
+    return 1
+
+
+def start_private_registry(ceph):
+    """Create and start the pricate registry on port 5000
+    Args:
+        ceph(ceph.ceph.Ceph): CephNode or list of CephNode object
+    """
+    ceph.exec_command(
```

Comment:
lets have generic method for this in - https://github.com/red-hat-storage/cephci/blob/master/cli/utilities/utils.py

Reply:
+1